scripts/autoscaling/aws/nodeup.py

```diff
@@ -543,8 +543,10 @@ def check_instance(ec2, ins_id, run_id, num_rep, time_rep):
     pipe_log('- Waiting for instance boot up...')
     result = poll_instance(sock, time_rep, ipaddr, port)
     rep = 0
-    while result != 0:
+    active = False
+    while result != 0 or not active:
```

Comment:
I think it shall be `and` instead of `or`

Reply:
No, or is relevant here, this statement check that instance return an answer on poll request and its state still active (Running or Pending), because some times we have situation when instance already terminating (according to status on a cloud), but instance its self still doing work and can answer on poll request

Reply:
So for the initial issue I added here checks in line `926` and `601`, and here is check for an other case, this case can be reproduced when you manually terminates instance on the aws console and at the same time nodeup already waits on polling

Reply:
OK, I got it